src/main/java/graphql/analysis/MaxQueryComplexityInstrumentation.java

```diff
@@ -6,32 +6,50 @@
 import graphql.execution.instrumentation.SimpleInstrumentation;
 import graphql.execution.instrumentation.parameters.InstrumentationValidationParameters;
 import graphql.validation.ValidationError;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.ArrayList;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
 
 import static graphql.Assert.assertNotNull;
 import static graphql.execution.instrumentation.SimpleInstrumentationContext.whenCompleted;
 
 /**
- * Prevents execution if the query complexity is greater than the specified maxComplexity
+ * Prevents execution if the query complexity is greater than the specified maxComplexity.
+ *
+ * Use the {@link Function<QueryComplexityInfo, Boolean>} parameter to supply a function to perform a custom action when the max complexity
+ * is exceeded. If the function returns {@code true} a {@link AbortExecutionException} is thrown.
  */
 @PublicApi
 public class MaxQueryComplexityInstrumentation extends SimpleInstrumentation {
 
+    private static final Logger log = LoggerFactory.getLogger(MaxQueryComplexityInstrumentation.class);
 
     private final int maxComplexity;
     private final FieldComplexityCalculator fieldComplexityCalculator;
+    private final Function<QueryComplexityInfo, Boolean> maxQueryComplexityExceededFunction;
 
     /**
      * new Instrumentation with default complexity calculator which is `1 + childComplexity`
      *
      * @param maxComplexity max allowed complexity, otherwise execution will be aborted
      */
     public MaxQueryComplexityInstrumentation(int maxComplexity) {
-        this(maxComplexity, (env, childComplexity) -> 1 + childComplexity);
+        this(maxComplexity, (queryComplexityInfo) -> true);
+    }
+
+    /**
+     * new Instrumentation with default complexity calculator which is `1 + childComplexity`
```

Comment:
Can we have some JavaDoc on the semantic of the function.  That is IF it returns true the query will be aborted

Reply:
The function documented on class level, similar as is done in e.g. `SimpleFieldValidation`. I can move/copy it to the constructor param if you want. Let me know.